bot/seasons/valentines/be_my_valentine.py

```diff
@@ -0,0 +1,235 @@
+            user = self.random_user(author, lovefest_role.members)
+            if user is None:
+                return await ctx.send("There are no users avilable to whome your valentine can be sent.")
+
+        embed = discord.Embed(
+            title=f'{emoji_1} {title} {user.display_name} {emoji_2}',
+            description=f'{valentine} \n **{emoji_2}From {ctx.author}{emoji_1}**',
+            color=Colours.pink
+        )
+        await channel.send(user.mention, embed=embed)
+
+    @commands.cooldown(1, 1800, BucketType.user)
+    @send_valentine.command(name='secret')
+    async def anonymous(self, ctx, user: typing.Optional[discord.Member] = None, *, valentine_type=None):
+        """
+        This command DMs a valentine to be given anonymous to a user if specified or a random user having lovefest role.
+
+        **This command should be DMed to the bot.**
+
+        syntax : .bemyvalentine secret [user](optional) [p/poem/c/compliment/or you can type your own valentine message]
+        (optional)
+
+        example : .bemyvalentine secret (sends valentine as a poem or a compliment to a random user in DM making you
+        anonymous)
+        example : .bemyvalentine secret Iceman#6508 p (sends a poem to Iceman in DM making you anonymous)
+        example : .bemyvalentine secret Iceman#6508 Hey I love you, wanna hang around ? (sends the custom message to
+        Iceman in DM making you anonymous)
+        """
+
+        if ctx.guild is not None:
+            # This command is only DM specific
+            msg = "You are not supposed to use this command in the server, DM the command to the bot."
+            return await ctx.send(msg)
+
+        if user:
+            if Lovefest.role_id not in [role.id for role in user.roles]:
+                message = f"You cannot send a valentine to {user} as he/she does not have the lovefest role!"
+                return await ctx.send(message)
+
+        if user == ctx.author:
+            # Well a user cant valentine himself/herself.
+            return await ctx.send('Come on dude, you cant send a valentine to yourself :expressionless:')
+
+        guild = self.bot.get_guild(id=Client.guild)
+        emoji_1, emoji_2 = self.random_emoji()
+        lovefest_role = discord.utils.get(guild.roles, id=Lovefest.role_id)
+        valentine, title = self.valentine_check(valentine_type)
+
+        if user is None:
+            author = ctx.author
+            user = self.random_user(author, lovefest_role.members)
+            if user is None:
+                return await ctx.send("There are no users avilable to whome your valentine can be sent.")
+
+        embed = discord.Embed(
+            title=f'{emoji_1}{title} {user.display_name}{emoji_2}',
+            description=f'{valentine} \n **{emoji_2}From anonymous{emoji_1}**',
+            color=Colours.pink
+        )
+        await ctx.author.send(f"Your message has been sent to {user}")
+        await user.send(embed=embed)
```

Comment:
Some users have DMs disabled in their privacy settings. This will result in the command failing to send anything, but the message indicates that it was sent successfully before the attempt happens.

Best thing to do here is to try sending it to the user first, and wrap a `try: except discord.Forbidden` around it to catch the case where it fails.

```py
        try:
            await user.send(embed=embed)
        except discord.Forbidden:
            await ctx.author.send(f"{user} has DMs disabled, so I couldn't send the message. Sorry!")
        else:
            await ctx.author.send(f"Your message has been sent to {user}")
```